### PR TITLE
Make it easier to use abilities as volatiles

### PR DESCRIPTION
**Pull request by urkerab**
Also making `Dex#getEffect` typesafe by moving the hack into the new `Battle#getVolatile`, which is allowed to return an `Effect` anyway.

### CODE REVIEW

**Comment by KrisXV** [comment]
Could you also utilize this in the Shared Power code?

**Comment by Zarel** [comment]
This doesn't actually solve the problem...

File: sim/battle.ts [hunk 1]
```diff
@@ -282,6 +282,13 @@ export class Battle {
 		this.prng = new PRNG(this.prng.startingSeed);
 	}
 
+	getVolatile(id: string): Effect {
+		if (id.startsWith('move:')) return Object.assign(Object.create(this.dex.getMove(id.slice(5))), {id});
```
Comment (scheibo) [inline sim/battle.ts:286]: This also still seems wrong, `id` for Effect should be an `ID`.

**Comment by urkerab** [comment]
The volatile system was laxly allowing you to use effect names as volatiles names. In order to support OMs I've had to switch everyone to use effect ids or prefixed move, item or ability ids. I've also removed support for using effects as volatile names, which was basically only linked volatiles at this point, so `volatiles[volatile].linkedStatus` is now a string rather than a PureEffect.

File: sim/battle.ts (hunk 1 above)
Comment (urkerab) [reply sim/battle.ts]: This was just bad naming.

**Comment by Zarel** [comment]
The correct thing, I think, is to replace `getRelevantEffects`?

**Comment by urkerab** [comment]
You mean `findEventHandlers`? How would that help? `addVolatile` and `removeVolatile` still need to know the real effect.

**Comment by Zarel** [comment]
You wouldn't use `addVolatile` and `removeVolatile` to add/remove items/abilities, of course.

**Comment by urkerab** [comment]
In that case, this has suffered scope creep...

**Comment by Zarel** [comment]
perhaps we _could_ do something, though

**Comment by Zarel** [comment]
I think it makes sense to support having non-IDs as keys in the volatiles table.

**Pull request closed without merging**